engine/Default/planet.inc

```diff
@@ -0,0 +1,25 @@
+<?php
+// Common code for all the planet display pages
+
+// If not on a planet, forward to current_sector.php
+if (!$player->isLandedOnPlanet()) {
+	if (USING_AJAX) {
+		// Auto-click current sector with javascript to avoid issues with ajax
+		// updates when the display page changes.
+		$container = create_container('skeleton.php', 'current_sector.php');
+		$currentSectorHREF = SmrSession::getNewHREF($container);
+		$template->addJavascriptForAjax('EVAL', "location.href = '$currentSectorHREF'");
```

Comment:
You should escape `'` in `$currentSectorHREF` - it shouldn't be a problem ever but it helps to future proof and defend against some side effects a bug could have

Reply:
Would you do this with something like this?
```php
$currentSectorHREF = str_replace("'", "\'", $currentSectorHREF);
```

Reply:
The safest would probably be to use `json_encode` in order to make sure it covers all the cases (eg `\'` would become `\\'` in the str_replace case)

Reply:
I see, so omitting the single quotes entirely (not even replacing them with double quotes)? I would have expected:
```php
$template->addJavascriptForAjax('EVAL', 'location.href = "' . json_encode($currentSectorHREF) . '"');
```

Sorry, I don't really know javascript... :/

Reply:
You should read up about JSON @hemberger, MDN is probably a good starting point: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/JSON
In short though it's just a human readable data format that is commonly used nowadays and has parsers/generators in basically every language.  Admittedly this is a slightly special usage of it in that it makes use of the fact that JSON is also valid javascript code that you can eval directly, rather than using the safer `JSON.parse`.  The main reason to do it this way is to use the existing mechanisms rather than adding an extra special `REDIRECT` element which would take just the string